keras/engine/training.py

```diff
@@ -910,6 +910,8 @@ def _fit_loop(self, f, ins, out_labels=None, batch_size=32,
                         for l, o in zip(out_labels, val_outs):
                             epoch_logs['val_' + l] = o
             callbacks.on_epoch_end(epoch, epoch_logs)
+            if do_validation:
+                callbacks.on_evaluate_end(epoch, epoch_logs)
```

Comment:
How is there any difference between this and `on_epoch_end`? In practice, both are called in succession, with nothing in between.